packages/tracing-operations/src/services/db/dbService.ts

```diff
@@ -296,11 +296,18 @@ export function dbServiceBuilder(db: DB) {
       }
     },
 
-    async deletePurposeErrors() {
+    async deletePurposesErrors(): Promise<void> {
       try {
-        return Promise.resolve();
+        const deletePurposesErrorsQuery = `
+          DELETE FROM tracing.purposes_errors pe
+          USING tracing.tracings t
+          WHERE pe.tracing_id = t.id
+          AND ((t.state = 'COMPLETED') 
+            OR (t.state = 'ERROR' AND pe.version < t.version));`;
```

Comment:
maybe we could avoid the state check and just go for the version?

Reply:
Yes can be simplified since COMPLETED state should have an higher version